manifests/master/gitea.pp

```diff
@@ -1,17 +1,53 @@
 # This assumes that the gitea rpm and dependencies have been cached by the
 # pltraining-bootstrap module.
 class classroom::master::gitea {
+
+  # Use public IP for ec2 hosted trainings
+  $gitea_address = $facts['ec2_metadata'] ? {
+        undef   => $ipaddress,
+        default => $facts['ec2_metadata']['public-ipv4']
+  }
+
   package { 'gitea':
     ensure => present,
-    before => File['/home/git/go/bin/custom/conf/app.ini'],
   }
 
-  file { '/home/git/go/bin/custom/conf/app.ini':
-    ensure  => file,
-    owner   => 'git',
-    group   => 'git',
-    mode    => '0644',
-    source  => 'puppet:///modules/classroom/app.ini',
+  ini_setting { "database user":
```

Comment:
It would be stupid if it were so, but do you know if the gitea app.ini file handles lowercase key names? They are all uppercase by default, but hope it can deal. Otherwise, all of the `setting` attribute values in the `ini_setting` resources need to be uppercased.